### ISSUE TITLE
Add certificate manager trust-configs for mTLS

### ISSUE THREAD

**Issue by imrannayer** [hashicorp/terraform-provider-google#14984]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment. If the issue is assigned to the "modular-magician" user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If the issue is assigned to a user, that user is claiming responsibility for the issue. If the issue is assigned to "hashibot", a community member has claimed the issue already.

### Description
[Certificate Manager Trust configs](https://cloud.google.com/certificate-manager/docs/how-it-works#trust-configs)
[API](https://cloud.google.com/certificate-manager/docs/reference/certificate-manager/rest/v1/projects.locations.trustConfigs)

### New or Affected Resource(s)

* google_certificate_manager_trust_config

### Potential Terraform Configuration

```tf
google_certificate_manager_trust_config "my_trust_config" {
  Project     = my-project-id
  description = "my description"
  name        = "my-test-trust-config"

  trust_stores {
    trust_anchors {
      pem_certificate = ""
    }

    intermediate_cas {
      pem_certificate = ""
    }

  }

}
```

### References

*

**Comment by FiskilBen** [comment]
Just wanted to check in that now a PR has been merged when should we expect to be able to use the `google_certificate_manager_trust_config` resource with one of the google terraform providers?

**Comment by imrannayer** [comment]
Fixed in #15562
